Makefile.dune

```diff
@@ -52,8 +65,16 @@ help:
 	@echo "  - 'dune build package.install' : build package in developer mode"
 	@echo "  - 'dune build -p package' : build package in release mode"
 	@echo ""
-	@echo " Packages _must_ be installed using release mode, use: 'dune install -p package'"
-	@echo " See Dune documentation for more information."
+	@echo " Packages _must_ be installed using release mode, to install a package use: "
```

Comment:
If a package is not built using the `release` profile then it may pick some dependencies from the current tree; so this is in general unsound and I don't want to support this use case.

Of course if you know what you are doing then you can install a package built in dev mode, in fact I do sometimes.

Reply:
"should" is probably more appropriate then. In English, "must" is a pretty strong word.

Reply:
`must` is the right word here; there is no case where installing without the profile is supported.

Reply:
>if you know what you are doing

That's "should" not "must" imo

Reply:
It is must as if you don't do it the installed package is broken [even if it works by chance]

Reply:
I'm not very convinced but I just noticed that the `_must_` is already in master so let's move on

Reply:
I can ask dune upstream if you want but I'm pretty sure they have said in the past that for installation `-p` is basically mandatory, otherwise you get random results depending what is in scope in your workspace.

Reply:
Yeah let's keep the must, if you are expert enough to understand `-p` then you know what you are doing. Actually missing `-p` has been a large source of bug reports in many packages [for example Nix or lablgtk] so that's why I'm so hesitant not to be categorical here.